File: apps/api/src/routes/health.ts
import type { FastifyInstance } from "fastify";
import { pool } from "../db/pool.js";
import { requireAuth } from "../middleware/auth.js";

export async function registerHealthRoutes(app: FastifyInstance) {
  app.get("/health", async () => ({ status: "ok", service: "medoffice-api" }));

  app.get("/api/health", async () => {
    try {
      await pool.query("SELECT 1");
      return { status: "ok", database: "connected" };
    } catch {
      return { status: "degraded", database: "disconnected" };
    }
  });

  app.get("/api/me", async (request, reply) => {
    const user = requireAuth(request);
    return {
      userId: user.userId,
      tenantId: user.tenantId,
      username: user.username,
      displayName: user.displayName,
      roleSlugs: user.roleSlugs,
      permissions: user.permissions,
      isAdmin: user.permissions.includes("admin:console"),
    };
  });
}

File: apps/server/internal/routes/health.go
package routes

import (
	"net/http"

	"github.com/gin-gonic/gin"
	"gorm.io/gorm"

	"medoffice/server/internal/auth"
)

// RegisterHealth 复刻 routes/health.ts：/health、/api/health、/api/me。
func RegisterHealth(r *gin.Engine, gdb *gorm.DB) {
	r.GET("/health", func(c *gin.Context) {
		c.JSON(http.StatusOK, gin.H{"status": "ok", "service": "medoffice-api"})
	})

	r.GET("/api/health", func(c *gin.Context) {
		if sqlDB, err := gdb.DB(); err == nil {
			if err := sqlDB.Ping(); err == nil {
				c.JSON(http.StatusOK, gin.H{"status": "ok", "database": "connected"})
				return
			}
		}
		c.JSON(http.StatusOK, gin.H{"status": "degraded", "database": "disconnected"})
	})

	r.GET("/api/me", auth.RequireAuth(), func(c *gin.Context) {
		u := auth.CurrentUser(c)
		c.JSON(http.StatusOK, gin.H{
			"userId":      u.UserID,
			"tenantId":    u.TenantID,
			"username":    u.Username,
			"displayName": u.DisplayName,
			"roleSlugs":   u.RoleSlugs,
			"permissions": u.Permissions,
			"isAdmin":     u.IsAdmin(),
		})
	})
}
